src/uct/ib/base/ib_md.c

```diff
@@ -219,6 +219,7 @@ typedef struct {
     void                          *address;
     size_t                        length;
     size_t                        first_mr_size;
+    int                           mr_idx;
```

Comment:
why needed? I would expect the thread will not exit with a partial registration: either it return UCS_OK, or returns an error and reverts all regisrtations

Reply:
yes. It is for the case where there is at least one failing thread. it will then be used to teardown the other threads that had succeeded.

Reply:
can we check this by the return value of the thread function, instead of adding this var to the context?

Reply:
we have to store per-thread because it's only when they are all joined that we know if one failed and we need to tear them down, so I think we cannot.

pthread_join could take return code and set it in `ctx` but seems better to have the thread update it itself.

Reply:
Right, we can wait untill all threads joined before deciding if need to clean up.
I think it's cleaner that "ctx" will only be used as input to the thread function, and not as both input and output. And the thread function can either fully succeed or fail (without "leftovers"), and the caller of the thread function will not need to check "ctx" internals in order to clean up. Similarly to what we do with regular (non-thread) functions.

Reply:
fixed